orchagent/intfsorch.cpp

```diff
@@ -1088,6 +1132,72 @@ void IntfsOrch::doTask(Consumer &consumer)
     }
 }
 
+void IntfsOrch::doSagTask(vector<FieldValueTuple> data, const string& op)
+{
+    if (op == SET_COMMAND)
+    {
+        for (auto idx : data)
+        {
+            string sag_mac_str = "";
+            const auto &field = fvField(idx);
+            const auto &value = fvValue(idx);
+            SWSS_LOG_NOTICE("process key %s, value %s", field.c_str(), value.c_str());
+            if (field == "gateway_mac")
+            {
+                sag_mac_str = value;
+            }
+
+            if (!sag_mac_str.empty())
+            {
+                MacAddress sag_mac;
+                try
+                {
+                    sag_mac = MacAddress(sag_mac_str);
+                }
+                catch (const std::invalid_argument &e)
+                {
+                    SWSS_LOG_ERROR("Invalid mac argument %s to %s()", sag_mac_str.c_str(), e.what());
+                    return;
+                }
+
+                IpPrefix linklocal_prefix = gRouteOrch->getLinkLocalEui64Addr(sag_mac);
+                const vector<sai_object_id_t>& vrf_ids = m_vrfOrch->getVRFids();
+
+                // add link local route for default vrf
+                gRouteOrch->addLinkLocalRouteToMe(gVirtualRouterId, linklocal_prefix);
```

Comment:
Should we add IP2Me route even if no vlan enables SAG?

Reply:
This is already handled by routeOrch, 
https://github.com/sonic-net/sonic-swss/blob/master/orchagent/routeorch.cpp#L143